### DISCUSSION TITLE
Gauge server crashed 5 times in the last 3 minutes the server will not be restarted

### DISCUSSION THREAD

**Discussion by jkrn1580** [getgauge/gauge#2276] [General]
As soon as we open VS code  and go to any spec file we see this crash error only in windows system, and we don't see Run spec, Scenario and debug options, but when we execute any spec from terminal using the command gauge run specs\.., then we see this below errors. can any one faced this issues, please help us in this.

Line 9: Unexpected reserved word
Line 3: Unexpected token
Line 3: Unexpected token     
Line 4: Unexpected identifier
Line 3: Unexpected token     
Line 1: Unexpected token
Line 3: Unexpected token        
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 3: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 5492: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1071: Unexpected token ...
Line 38571: Unexpected token ...
Line 746: Unexpected token ...
Line 45: Unexpected token ...
Line 596: Unexpected token
Line 1: Unexpected token ILLEGAL
EISDIR: illegal operation on a directory, read
Error: EISDIR: illegal operation on a directory, read
    at Object.readSync (fs.js:614:3)
    at tryReadSync (fs.js:383:20)
    at Object.readFileSync (fs.js:420:19)
    at C:\Users\Jagadeesh.Rangapuram\AppData\Roaming\gauge\plugins\js\2.3.17\src\static-loader.js:66:28
    at Array.forEach (<anonymous>)
    at Object.loadFiles [as load] (C:\Users\{user}\AppData\Roaming\gauge\plugins\js\2.3.17\src\static-loader.js:65:40)
    at C:\Users\{user}\AppData\Roaming\gauge\plugins\js\2.3.17\src\gauge.js:32:12
Updates are available. Run `gauge update -c` for more info.
Error ----------------------------------

[Gauge]
Error: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0xc0005eb240, 0xed1480, 0x1492a30)
        C:/hostedtoolcache/windows/go/1.16.13/x64/src/runtime/debug/stack.go:24 +0xa5
main.recoverPanic()
        D:/a/gauge/gauge/gauge.go:27 +0x65
panic(0xed1480, 0x1492a30)
        C:/hostedtoolcache/windows/go/1.16.13/x64/src/runtime/panic.go:965 +0x1c7
github.com/getgauge/gauge/runner.(*GrpcRunner).ExecuteMessageWithTimeout(0x0, 0xc0000f0000, 0x0, 0xc0004841c0, 0x0)
        D:/a/gauge/gauge/runner/grpcRunner.go:195 +0x29
github.com/getgauge/gauge/validation.(*SpecValidator).validateStep(0xc000484000, 0xc00000ef00, 0xc000285040, 0xf)
        D:/a/gauge/gauge/validation/validate.go:345 +0x1ba
github.com/getgauge/gauge/validation.(*SpecValidator).Step(0xc000484000, 0xc00000ef00)
        D:/a/gauge/gauge/validation/validate.go:313 +0x5c5
github.com/getgauge/gauge/validation.(*SpecValidator).Step(0xc000484000, 0xc000185400)
        D:/a/gauge/gauge/validation/validate.go:307 +0x87
github.com/getgauge/gauge/gauge.(*Specification).Traverse(0xc0000fa540, 0x10ab7f8, 0xc000484000, 0xc0005eb988)
        D:/a/gauge/gauge/gauge/specification.go:261 +0x26a
github.com/getgauge/gauge/validation.(*SpecValidator).validate(0xc000484000, 0xc000484000, 0x13, 0x100010000)
        D:/a/gauge/gauge/validation/validate.go:298 +0x94
github.com/getgauge/gauge/validation.(*validator).Validate(0xc0005ebb38, 0x10aad00)
        D:/a/gauge/gauge/validation/validate.go:284 +0x11c
github.com/getgauge/gauge/validation.ValidateSpecs(0xc00020a0c0, 0x1, 0x1, 0x0, 0x0)
        D:/a/gauge/gauge/validation/validate.go:174 +0x245
github.com/getgauge/gauge/execution.glob..func1(0xc00020a0c0, 0x1, 0x1, 0x0)
        D:/a/gauge/gauge/execution/execute.go:107 +0x119
github.com/getgauge/gauge/cmd.execute(0x14a0760, 0xc00020a0c0, 0x1, 0x1)
        D:/a/gauge/gauge/cmd/run.go:241 +0x22e
github.com/getgauge/gauge/cmd.glob..func12(0x14a0760, 0xc00020a0c0, 0x1, 0x1)
        D:/a/gauge/gauge/cmd/run.go:90 +0x26e
github.com/spf13/cobra.(*Command).execute(0x14a0760, 0xc00020a090, 0x1, 0x1, 0x14a0760, 0xc00020a090)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x149e6e0, 0x0, 0xc000061f78, 0xc0000a3f78)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/getgauge/gauge/cmd.Parse(0x7bb901, 0x0)
        D:/a/gauge/gauge/cmd/cmd.go:112 +0x45
main.main()
        D:/a/gauge/gauge/gauge.go:19 +0x4f

Get Support ----------------------------
        Docs:          https://docs.gauge.org
        Bugs:          https://github.com/getgauge/gauge/issues
        Chat:          https://github.com/getgauge/gauge/discussions

Your Environment Information -----------
        windows, 1.4.3, f98dd40
        html-report (4.1.4), js (2.3.17), screenshot (0.1.0), spectacle (0.1.4), xml-report (0.2.3)

Reagards
Jagadeesh

**Comment by jkrn1580** [comment]
can some one please help on this issue.

**Comment by zabil** [comment]
Please share the version of node you are using. Please make sure that it's node 16+

**Reply by jkrn1580** [reply]
Thanks Zabil, please find the node versions.
![image](https://user-images.githubusercontent.com/45527086/185649573-5f1b9d39-47ad-4608-8e70-7f067746d951.png)

Please note , have updated Node version to 16.12.0 and we see now Run Scenario and Debug Scenario options in spec file, but when we execute scenarios we see these below errors, this happens only in Windows System, Mac its working fine. kindly do suggest if we are missing something. thanks

--hide-suggestion
Line 9: Unexpected reserved word
Line 3: Unexpected token
Line 3: Unexpected token
Line 4: Unexpected identifier
Line 3: Unexpected token
Line 1: Unexpected token
Line 3: Unexpected token
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 3: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 5492: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1071: Unexpected token ...
Line 38571: Unexpected token ...
Line 746: Unexpected token ...
Line 45: Unexpected token ...
Line 596: Unexpected token
Line 1: Unexpected token ILLEGAL
Line 39: Unexpected token ...
Line 6: Illegal return statement
Line 1: Unexpected string
Line 218: Unexpected token ...
Line 42: Unexpected token ...
Line 33: Unexpected token ...
Line 941: Unexpected token ...
Line 471: Unexpected token ...
Line 1: Unexpected token ILLEGAL
Line 5: Unexpected token ...
Line 26: Unexpected token ...
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 14: Unexpected token ...
Line 1: Unexpected token ILLEGAL
Line 3: Unexpected token
Line 2: Unexpected identifier
Line 3: Unexpected token {
Line 7: Unexpected token ...
Line 1: Unexpected token ILLEGAL
Line 1519: Unexpected token
Line 1: Unexpected token ILLEGAL
Line 3: Unexpected identifier
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 333: Unexpected token
Line 2: Unexpected token
Line 2: Unexpected token
Line 2: Unexpected token
Line 10: Unexpected token
Line 1: Unexpected token ILLEGAL
Line 120: Unexpected token ...
Line 95: Unexpected token ...
Line 13: Unexpected token {
Line 47: Unexpected token {
Line 43: Unexpected token {
Line 22: Unexpected token {
Line 62: Unexpected token {
Line 5: Unexpected token ...
Line 8: Unexpected token ...
Line 293: Unexpected token {
{"logLevel":"info","message":{}}

Line 10: Unexpected token
Line 10: Unexpected token
Line 26: Unexpected token ...
Line 1: Unexpected token
Line 163: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 12: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token ILLEGAL
Line 64: Unexpected token ...
Line 5: Unexpected token ...
Line 20: Unexpected token ...
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 89: Unexpected token ...
Line 20: Unexpected token ...
Line 295: Unexpected token ...
Line 7: Unexpected identifier
Line 94: Unexpected token ...
Line 1: Unexpected token
Line 32: Unexpected token
Line 1: Unexpected token
Line 2: Unexpected token
Line 3: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 622: Unexpected token
Line 2: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 68: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 10: Unexpected token
Line 18: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 2: Unexpected token
Line 11: Illegal return statement
Line 11: Illegal return statement
Line 346: Unexpected token ...
Line 106: Unexpected token ...
Line 49: Unexpected token ...
Line 21: Unexpected token ...
Line 7: Unexpected token ...
Line 25: Unexpected token {
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 1: Illegal return statement
Line 39: Unexpected token ...
{"logLevel":"info","message":{}}

{"logLevel":"info","message":{}}

{"logLevel":"info","message":{}}

{"logLevel":"info","message":{}}

Line 1: Unexpected token
Line 17: Unexpected token ...
Line 43: Unexpected token ...
Line 29: Unexpected token ...
Line 26: Unexpected token ...
Line 30: Unexpected token ...
Line 28: Unexpected token ...
Line 13: Unexpected token ...
Line 14: Unexpected token ...
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 34: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 21: Unexpected token
Line 3: Unexpected token
Line 77: Unexpected token ...
Line 6: Unexpected token ...
Line 50: Unexpected token ...
Line 62: Unexpected token ...
Line 62: Unexpected token ...
Line 44: Unexpected token ...
Line 12: Unexpected token ...
Line 10: Unexpected token ...
Line 14: Unexpected token ...
Line 234: Unexpected token ...
Line 146: Unexpected token ...
Line 11: Unexpected token ...
Line 89: Unexpected token ...
Line 123: Unexpected token
Line 1: Unexpected token ILLEGAL
Line 10: Unexpected token
Line 10: Unexpected token
Line 29: Unexpected token ...
Line 30: Unexpected token ...
Line 29: Unexpected token {
Line 28: Unexpected token ...
Line 1: Unexpected token ILLEGAL
Line 1: Unexpected token
Line 163: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 12: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token ILLEGAL
{"logLevel":"info","message":{}}

{"logLevel":"info","message":{}}

{"logLevel":"info","message":{}}

Line 1: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 2: Unexpected token
Line 2: Unexpected token
Line 1: Unexpected token
Line 1: Unexpected token
Line 25: Unexpected token ...
Line 1: Unexpected token
Line 24: Unexpected token
{"logLevel":"info","message":{}}

Line 349: Unexpected token ...
Line 68: Unexpected token ...
Line 66: Unexpected token ...
Line 414: Unexpected token ...
Line 29: Unexpected token ...
Line 30: Unexpected token ...
Line 28: Unexpected token ...
Line 5: Unexpected token
Line 9: Unexpected token
Line 6: Unexpected token
Line 7: Unexpected token
Line 6: Unexpected token
Line 6: Unexpected token
